Core/Object Arts/Dolphin/IDE/Base/SmalltalkSystem.cls

```diff
@@ -1888,6 +1888,12 @@ hierarchyBrowserClass: toolClass
 
 	hierarchyBrowserClass := toolClass!
 
+iconForMethod: aCompiledMethod
+	"Answer an icon that can be used to represent the <CompiledMethod> argument in a development environment."
+
+	aCompiledMethod isDeprecated ifTrue: [^CompiledMethod deprecatedIcon].
```

Comment:
I think you should move the CompiledMethod class>>deprecatedIcon|privateIcon|publicIcon methods to SmalltalkSystem too. They are private, and there isn't really a good reason to override these in the CompiledMethod class hierarchy. I realise this will mean replacing the refs in StsMethodEdition, but it is already dependent on the Development System package.